syntax = "proto3";

package masterworker;

service AssignTask {
    rpc Ping(PingRequest) returns (TaskReply){}
    rpc Map(MapRequest) returns (TaskReply){}
    rpc Reduce(ReduceRequest) returns (TaskReply){}
}

message ShardPiece{
    string file_name = 1;
    int32 start_index = 2;
    int32 end_index = 3;
}

message PingRequest{}

message MapRequest{
    string user_id = 1;
    string job_id = 2;
    repeated ShardPiece shard = 3;
	int32 num_reducers = 4;
}

message ReduceRequest{
    string user_id = 1;
    string output_file = 2;
    string job_id = 3;
    repeated string input_files = 4;
    string section = 5;
}

message TaskReply{
    string job_id = 1;
    string out_file = 2;
    string task_type = 3;
}
